aggsender/statuschecker/initial_state_test.go

```diff
@@ -190,57 +206,115 @@ func TestRegularCases(t *testing.T) {
 			localCert:       &certTestData{hash2, 2, agglayertypes.Proven},
 			agglayerSettled: &certTestData{hash1, 3, agglayertypes.Proven},
 			agglayerPending: nil,
-			resultActions:   &initialStateResultTest{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 3, agglayertypes.Proven}},
+			resultActions:   []*initialStateResultTest{{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 3, agglayertypes.Proven}}},
 		},
 		{
-			name:            "12|ID2, h2 , NA		| ID1, h2 , settled		| ID1, h3 , !=inError           |  store(PENDING)",
+			name:            "12|ID2, h2 , NA		| ID1, h2 , settled		| ID1, h3 , !=inError           |  store(PENDING)&store(SETTLED)",
 			localCert:       &certTestData{hash2, 2, agglayertypes.Proven},
 			agglayerSettled: &certTestData{hash1, 2, agglayertypes.Settled},
 			agglayerPending: &certTestData{hash1, 3, agglayertypes.Proven},
-			resultActions:   &initialStateResultTest{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 3, agglayertypes.Proven}},
+			resultActions: []*initialStateResultTest{
+				{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 3, agglayertypes.Proven}},
+				{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 2, agglayertypes.Settled}},
+			},
 		},
 		{
-			name:            "13|ID2, h2 , NA		| ID1, h2 , settled		| ID1, h3 , inError             | store(PENDING)",
-			localCert:       &certTestData{hash2, 2, agglayertypes.Proven},
-			agglayerSettled: &certTestData{hash1, 2, agglayertypes.Settled},
-			agglayerPending: &certTestData{hash1, 3, agglayertypes.InError},
-			resultActions:   &initialStateResultTest{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 3, agglayertypes.InError}},
+			name:             "13|ID2, h2 , NA		| ID1, h2 , settled		| ID1, h3 , inError             | store(PENDING)&store(SETTLED)",
+			localCert:        &certTestData{hash2, 2, agglayertypes.Proven},
+			localSettledCert: nil,
+			agglayerSettled:  &certTestData{hash1, 2, agglayertypes.Settled},
+			agglayerPending:  &certTestData{hash1, 3, agglayertypes.InError},
+			resultActions: []*initialStateResultTest{
+				{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 3, agglayertypes.InError}},
+				{InitialStatusActionInsertNewCert, "", &certTestData{hash1, 2, agglayertypes.Settled}},
+			},
+		},
+		{
+			name:             "14| LocalCert: ID3, h1, pending	| LocalSettled: ID2, h1	| AgglayerSettled: ID1, h2   | AgglayerPending: ID3, h1 | agglayer doesn't have settled cert",
+			localCert:        &certTestData{hash1, 3, agglayertypes.Pending},
+			localSettledCert: &certTestData{hash1, 2, agglayertypes.Settled},
+			agglayerSettled:  &certTestData{hash2, 1, agglayertypes.Settled},
+			agglayerPending:  &certTestData{hash1, 3, agglayertypes.Pending},
+			resultError:      true,
+		},
+		{
+			name:             "14| LocalCert: ID3, h1, pending | LocalSettled: ID2, h1 | AgglayerSettled: ID2, h1 | AgglayerPending: ID3, h1 | store(PENDING) & none(SETTLED)",
```

Comment:
Should this be test case #15 (and increment each subsequent by one), or is deliberately this way?

```suggestion
			name:             "15| LocalCert: ID3, h1, pending | LocalSettled: ID2, h1 | AgglayerSettled: ID2, h1 | AgglayerPending: ID3, h1 | store(PENDING) & none(SETTLED)",
```

Reply:
Fix: https://github.com/agglayer/aggkit/pull/753/commits/7ab140b26d79ace0bfe83b72ead1c4ed52dafcc8